test: update server type list index to ignore cx11 entry

apricote opened the pull request:
Do not hardcode the list indices of the two server types we verify. Instead get the list from the API and calculate the expected indices.

This makes us more resilient against changes to the list when server types are deprecated/removed.

apricote committed:
test(server-type): use dynamic index to be resilient against list changes

Do not hardcode the list indices of the two server types we verify.
Instead get the list from the API and calculate the expected indices.

This makes us more resilient against changes to the list when server
types are deprecated/removed.

jooola commented:
Could we update the test data to use cx11 instead? And fix it again once we removed it?

apricote commented:
> Could we update the test data to use cx11 instead? And fix it again once we removed it?

I updated the PR to instead get the indices of `cpx11` and `cpx21` and use those instead of hardcoding `0` and `1`. What do you think about that approach?

jooola commented on internal/servertype/data_source_test.go:
```diff
@@ -67,6 +73,28 @@ func TestAccDataSourceList(t *testing.T) {
+	}
+
+	indexCPX11 := slices.IndexFunc(serverTypes, func(serverType *hcloud.ServerType) bool { return serverType.Name == "cpx11" })
+	indexCPX21 := slices.IndexFunc(serverTypes, func(serverType *hcloud.ServerType) bool { return serverType.Name == "cpx21" })
```
I think this is overkill, if you really want to keep this, please extract it in a function so we don't make this test more complex that it needs to be.

jooola replied:
I think updating the code once cx11 is really out is better.

jooola committed:
test: fix server-type list index

jooola commented:
Let me know what you think

apricote commented:
Works for me!
